tiledb/api/src/array/dimension/mod.rs

```diff
@@ -462,10 +191,30 @@ impl Builder {
         let constraints = constraints.into();
         constraints.verify_type_compatible(datatype)?;
 
-        let c_datatype = datatype.capi_enum();
+        let c_datatype = ffi::tiledb_datatype_t::from(datatype);
         let c_name = cstring!(name);
-        let c_domain = constraints.domain_ptr();
-        let c_extent = constraints.extent_ptr();
+        let c_domain = dimension_constraints_go!(
```

Comment:
This was the body of `DimensionConstraints::domain_ptr`.
It didn't really feel right to have an otherwise Rust-only crate provide methods which deal with raw pointers.